packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -512,7 +512,9 @@ class WebBuiltInAssets extends Target {
   List<String> get depfiles => const <String>[];
 
   @override
-  List<Source> get inputs => const <Source>[];
+  List<Source> get inputs => const <Source>[
+    Source.hostArtifact(HostArtifact.flutterWebSdk),
+  ];
```

Comment:
Yeah, it would be slightly better with outputs, as I think it would be able to invalidate if the user manually tinkered with the output files. Maybe we could do that as a followup at some point.